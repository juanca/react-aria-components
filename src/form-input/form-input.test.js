import React from 'react';
import renderer from 'react-test-renderer';
import {
  render,
  screen,
} from '@testing-library/react';
import FormInput from './form-input.js';

describe('<FormInput />', () => {
  const requiredProps = {
    id: 'test-id',
    label: 'Test label',
  };

  it('has default', () => {
    expect(renderer.create(<FormInput {...requiredProps} />).toJSON()).toMatchSnapshot();
  });

  describe('id API', () => {
    it('can be set', () => {
      render(<FormInput {...requiredProps} id="unique-id" />);
      expect(screen.getByLabelText('Test label')).toBeDefined();
    });
  });

  describe('label API', () => {
    it('can be set', () => {
      render(<FormInput {...requiredProps} label="Unique label" />);
      expect(screen.getByLabelText('Unique label')).toBeDefined();
    });
  });
});
