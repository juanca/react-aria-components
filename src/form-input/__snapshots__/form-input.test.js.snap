// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormInput /> has default 1`] = `
Array [
  <label
    htmlFor="test-id"
  >
    Test label
  </label>,
  <input
    id="test-id"
  />,
]
`;
